contrib/lua/premake5.lua

```diff
@@ -21,8 +21,8 @@ project "lua-lib"
 		"etc/*.c"
 	}
 
-	configuration "linux or bsd or hurd or aix"
+	filter "system:linux or system:bsd or system:hurd or system:aix"
```

Comment:
> > do you really need to put "not" between the scope and the keyword

> Currently yes. I'm not opposed to supporting "not system:windows" in addition to the current syntax though.

Sorry I'm barely keeping up lately, but I spotted this one while skimming... I'm not sure I can get behind this.
It's not clear to me what `"not system:windows"` means, whereas I find `"system: not windows"` to be quite obvious.
I fear that supporting both possibilities will lead to confusion, and the latter is much more intuitive to my mind.
What's the advantage of supporting the former expression?